frontend/src/components/LeaguesDialog.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState } from 'react';
+import { Box, Button, styled, TextField } from '@mui/material';
+import { leagueDataSource } from '../datasources/index';
+import { DialogType } from '../datasources/LeagueDataSource';
+
+const PopUpContainer = styled(Box)({
+    height: "fit-content",
+    padding: "15px",
+    display: "flex", 
+    alignItems: "center",
+    flexDirection: "column",
+    gap: "10px"
+});
+
+const CustomButton = styled(Button)({
+    backgroundColor: " #E5B226",
+    color: "#FBF9F9",
+    boxShadow: "0px 3px #D9D9D9",
+    borderRadius: "10px",
+    fontWeight: "bold",
+    ":hover":{
+        backgroundColor: "rgb(140, 109, 23)",
+        boxShadow: "none"
+    }
+});
+
+interface DialogProps {
+    dialogType: DialogType;
+    onClose : (value: boolean) => void;
+    onUpdate : () => void;
+};
+
+export function LeaguesDialog(props: DialogProps){
+
+    const [errorMessage, setErrorMessage] = useState<string>("");
+    const [userInput, setUserInput] = useState<string>("");
+
+    const isCreateDialog: boolean = props.dialogType === DialogType.CREATE;
+        
+    async function handleAddLeagueToBackend() {// changer de nom
```

Comment:
gnagna et le commentaire ahah ! 
```suggestion
    async function updateLeagues() {
```